### PR TITLE
fix pytket 1.35 issues

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Generating the QIR with pytket 1.35. gives for some circuits differences compared to the previous pytket version.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: tests/qir/test_pytket_qir_14-QIRProfile.PYTKET.ll [hunk 1]
```diff
@@ -18,9 +18,9 @@ entry:
   %1 = call i1* @create_creg(i64 10)
   %2 = call i1* @create_creg(i64 10)
   %3 = call i1* @create_creg(i64 9)
-  %4 = call i1* @create_creg(i64 64)
-  %5 = call i1* @create_creg(i64 64)
-  %6 = call i1* @create_creg(i64 64)
+  %4 = call i1* @create_creg(i64 8)
```
Comment (cqc-melf) [inline tests/qir/test_pytket_qir_14-QIRProfile.PYTKET.ll:21]: I am a bit confused by the size of the `c"tk_SCRATCH_BITREG_0\00"` and other, they are still split, but only into 8 bits each? This looks like a bug to me?

File: tests/qir/test_pytket_qir_14-QIRProfile.ADAPTIVE.ll [hunk 2]
```diff
@@ -17,17 +17,18 @@ entry:
   br i1 false, label %condb0, label %contb0
 
 condb0:                                           ; preds = %entry
+  call void @__quantum__qis__x__body(%Qubit* null)
```
Comment (cqc-alec) [inline tests/qir/test_pytket_qir_14-QIRProfile.ADAPTIVE.ll:20]: Do you know why these changes are happening?

Comment (cqc-melf) [reply tests/qir/test_pytket_qir_14-QIRProfile.ADAPTIVE.ll:20]:
```
    c = Circuit(1, name="test_classical")
    a = c.add_c_register("a", 8)
    b = c.add_c_register("b", 10)
    d = c.add_c_register("d", 10)

    c.add_c_setbits([True], [a[0]])
    c.add_c_setbits([False, True] + [False] * 6, list(a))
    c.add_c_setbits([True, True] + [False] * 8, list(b))

    c.add_c_setreg(23, a)
    c.add_c_copyreg(a, b)

    c.add_classicalexpbox_register(a + b, d)  # type: ignore
    c.add_classicalexpbox_register(a - b, d)  # type: ignore
    # c.add_classicalexpbox_register(a * b // d, d)
    c.add_classicalexpbox_register(a << 1, a)  # type: ignore
    c.add_classicalexpbox_register(a >> 1, b)  # type: ignore

    c.X(0, condition=reg_eq(a ^ b, 1))
    c.X(0, condition=(a[0] ^ b[0]))
    c.X(0, condition=reg_eq(a & b, 1))
    c.X(0, condition=reg_eq(a | b, 1))

    c.X(0, condition=a[0])
    c.X(0, condition=reg_neq(a, 1))
    c.X(0, condition=if_not_bit(a[0]))
    c.X(0, condition=reg_gt(a, 1))
    c.X(0, condition=reg_lt(a, 1))
    c.X(0, condition=reg_geq(a, 1))
    c.X(0, condition=reg_leq(a, 1))
    c.Phase(0, condition=a[0])

Comment (cqc-melf) [reply tests/qir/test_pytket_qir_14-QIRProfile.ADAPTIVE.ll:20]: Above is the testcase for this QIR files, from my understanding the quantum gates get reordered, but I don't know why. From my understanding this is a generating an equivalent circuit.

Comment (cqc-alec) [reply tests/qir/test_pytket_qir_14-QIRProfile.ADAPTIVE.ll:20]: Oh I see, yes, reordering I can understand because of the change in classical expression handling.

**Review by cqc-alec — APPROVED** [review]
